examples/hackweek_demo/acquire_data.py

```diff
@@ -0,0 +1,130 @@
+"""Script to format on-chain hyperdrive pool, config, and transaction data post-processing"""
+from __future__ import annotations
+
+# std lib
+import logging
+import time
+
+# third party
+from eth_typing import BlockNumber, URI
+from web3 import Web3
+import pandas as pd
+
+# custom
+from extract_data_logs import calculate_spot_price
+from elfpy.data import contract_interface
+from elfpy.utils import outputs as output_utils
+
+# pylint: disable=too-many-arguments, invalid-name
+
+
+def prepare_data(trades, pool):
+    """Prepare data for analysis."""
+    # remove all empty values from dicts
+    trades = {k: v for k, v in trades.items() if v}
+    pool = {k: v for k, v in pool.items() if v}
+    # turn to dataframes
+    pool_df = pd.DataFrame(pool).T
+    pool_df["shareReserves"] = pool_df["shareReserves"].astype(float)
+    pool_df["bondReserves"] = pool_df["bondReserves"].astype(float)
+    pool_df["lpTotalSupply"] = pool_df["lpTotalSupply"].astype(float)
+    pool_df["sharePrice"] = pool_df["sharePrice"].astype(float)
+    pool_df["timestamp"] = pool_df["timestamp"].iloc[0] + pool_df["blockNumber"] * 12
+    pool_df["spot_price"] = calculate_spot_price(
+        share_reserves=pool_df["shareReserves"],
+        bond_reserves=pool_df["bondReserves"],
+        lp_total_supply=pool_df["lpTotalSupply"],
+    )
+    txns, logs, receipts = [], [], []
+    for block in trades.values():
```

Comment:
it looks like you can take out all the logs stuff from this function.  this function is doing 2 things and usually better to keep them singly focused.  I'd make another function to handle the logs.